cddl_gen/cddl_gen.py

```diff
@@ -196,23 +200,34 @@ def from_cddl(cddl_class, cddl_string, default_max_qty, *args, **kwargs):
         my_types = dict()
 
         type_strings = cddl_class.get_types(cddl_string)
-        my_types = {my_type: None for my_type in type_strings.keys()}
+        my_types = \
+            {my_type: None for my_type, val in type_strings.items() if not val.startswith("&(")}
+        my_control_groups = \
+            {my_cg: None for my_cg, val in type_strings.items() if val.startswith("&(")}
 
         # Parse the definitions, replacing the each string with a
         # CodeGenerator instance.
         for my_type, cddl_string in type_strings.items():
-            parsed = cddl_class(*args, default_max_qty, my_types, **kwargs, base_name=my_type)
-            parsed.get_value(cddl_string.replace("\n", " "))
-            my_types[my_type] = parsed.flatten()[0]
-            my_types[my_type].set_id_prefix("")
+            parsed = cddl_class(*args, default_max_qty, my_types, my_control_groups, **kwargs,
+                                base_name=my_type)
+            parsed.get_value(cddl_string.replace("\n", " ").lstrip("&"))
+            parsed = parsed.flatten()[0]
+            if my_type in my_types:
+                my_types[my_type] = parsed
+            elif my_type in my_control_groups:
+                my_control_groups[my_type] = parsed
 
         counter(True)
 
         # post_validate all the definitions.
         for my_type in my_types:
+            my_types[my_type].set_id_prefix("")
             my_types[my_type].post_validate()
+        for my_control_group in my_control_groups:
+            my_control_groups[my_type].set_id_prefix("")
```

Comment:
> Pycharm is telling me that my_type might be referenced before it is initialized here.

Yes, shouldn't this be `my_control_groups[my_control_group].set_id_prefix('')`?

Reply:
Also, why isn't the ID prefix just defaulted to `''`, instead of having to explicitly set it to an empty string every time?